### PR TITLE
feat: Datasts graph provisioning process

### LINKED ISSUE #1257: KG API to look for Datasets in the Datasets graph
As a KG API user, I'd like searching for Datasets is fast.

**The problem**
Currently, finding a Dataset in the TS is difficult and slow. There are a number of reasons for that, where the most significant are:
* huge amount of data due to the way Datasets are modelled in the TS;
* no single place where the data can be found; the search has to go through all Project graphs

**The idea:**
It looks like creating a dedicated Datasets graph where only findable/discoverable Datasets should be kept might be a good solution. Such a graph would keep info about Datasets that are leaves of either the import or the derivation graphs for each Dataset. Additionally, each Dataset in the proposed graph would need to keep a link to a Project (or Projects) where it belongs so, in case more information is needed, a quick jump to the graph with full Dataset info can be done. It's worth mentioning, the new Datasets graph would not contain the full info about each Dataset but only the ones needed for the search queries.

**Acceptance criteria:**

- [x] transformation flow changes
  - [x] what we need to keep in the Datasets graph (for each DS)?
    - [x] resourceId, name, visibility (the most visible), date, creators, keywords, description, images, links to projectIds
  - [x] how we compute the data?
    - [x] for each DS in the payload, collect the last modification (except the last modification is invalidation)
    - [x] all invalidated DS should be removed from the new graph (either remove the DS from the graph or just a link to a project if there are more than one)
    - [x] the change to the new graph needs to be done in an upsert way
  - [x] how we put the data to the graph?
    - [x] create a new module with read and write APIs; both TG and KG to use the API
- [x] how do we put the data for all the Datasets we've got in the TS?
- [x] Cross-Enitity Search API changes
- [ ] Make sure the UI can work with the new way Cross-Enitity Search returns info/links about Datasets (topmostSameAs instead of identifier)
- [ ] ~Free-Text DS Seach API changes~ - we don't have to do it as this API will be removed in the future.

resolves #1077

### PR DESCRIPTION
**Pull request by jachro**
This PR adds a new process to provision a new `Datasets` graph in the TS. The new graph will contain only Datasets that should be discoverable for the search APIs. Additionally, it will contain only a subset of Dataset entity properties.

/deploy #persist

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -39,13 +58,25 @@ lazy val generators = Project(
...
+)
+
 lazy val tinyTypes = Project(
   id = "tiny-types",
   base = file("tiny-types")
 ).settings(
   commonSettings
 ).dependsOn(
-  generators % "test->test"
+  triplesStoreClient % "compile->compile",
```
Comment (eikek) [inline build.sbt]: sbt can combine the scope mappings in a single string like `% "compile->compile; test->test"` - but totally fine like this as well, ofc

File: entities-search/src/test/scala/io/renku/entities/searchgraphs/commands/QuadsEncodersSpec.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.entities.searchgraphs.commands
+
+import org.scalatest.matchers.should
+import org.scalatest.wordspec.AnyWordSpec
+
+class QuadsEncodersSpec extends AnyWordSpec with should.Matchers {}
```
Comment (eikek) [inline entities-search/src/test/scala/io/renku/entities/searchgraphs/commands/QuadsEncodersSpec.scala]: Is this intentionally empty?

File: renku-model/src/main/scala/io/renku/graph/model/entities/Dataset.scala [hunk 3]
```diff
@@ -499,41 +499,64 @@ object Dataset {
...
-        DataProperty(renku / "originalIdentifier", xsd / "string"),
-        DataProperty(prov / "invalidatedAtTime", xsd / "dateTime"),
-        DataProperty(schema / "description", xsd / "string"),
-        DataProperty(schema / "keywords", xsd / "string"),
-        DataProperty(schema / "license", xsd / "string"),
-        DataProperty(schema / "version", xsd / "string")
-      ),
-      ReverseProperties(PublicationEvent.ontology)
-    )
+  object Ontology {
```
Comment (eikek) [inline renku-model/src/main/scala/io/renku/graph/model/entities/Dataset.scala:502]: ❤️

File: renku-model-tiny-types/src/main/scala/io/renku/graph/model/images/Image.scala [hunk 4]
```diff
@@ -54,9 +54,16 @@ object Image {
     } yield Image(resourceId, uri, position)
   }
 
-  val ontology: Type = Type.Def(
-    Class(schema / "ImageObject"),
-    DataProperty(schema / "contentUrl", xsd / "string"),
-    DataProperty(schema / "position", xsd / "int")
-  )
+  object Ontology {
```
Comment (eikek) [inline renku-model-tiny-types/src/main/scala/io/renku/graph/model/images/Image.scala:57]: Thanks!

**Review by eikek — APPROVED** [review]
Very nice! 🚀

File: build.sbt (hunk 1 above)
Comment (jachro) [reply build.sbt]: Yes, I didn't know you can do it like this but saw it on your PR. I'll change it here.

File: entities-search/src/test/scala/io/renku/entities/searchgraphs/commands/QuadsEncodersSpec.scala (hunk 2 above)
Comment (jachro) [reply entities-search/src/test/scala/io/renku/entities/searchgraphs/commands/QuadsEncodersSpec.scala]: This bit PRs... I simply forgot about adding tests :)

Comment (jachro) [reply entities-search/src/test/scala/io/renku/entities/searchgraphs/commands/QuadsEncodersSpec.scala]: Wrong, actually, not the tests are in the `EncodersSpec`

**Pull request merged**